rclcpp_components/src/component_container_isolated.cpp

```diff
@@ -13,37 +13,57 @@
 // limitations under the License.
 
 #include <memory>
-#include <vector>
 #include <string>
+#include <vector>
 
 #include "rclcpp/rclcpp.hpp"
 #include "rclcpp/utilities.hpp"
 #include "rclcpp_components/component_manager_isolated.hpp"
 
 int main(int argc, char * argv[])
 {
-  /// Component container with dedicated single-threaded executors for each components.
+  /// Component container with dedicated executor for each component
   rclcpp::init(argc, argv);
+
   // parse arguments
-  bool use_multi_threaded_executor{false};
+  // valid entries: --executor-type single-threaded, --executor-type multi-threaded, --executor-type events
   std::vector<std::string> args = rclcpp::remove_ros_arguments(argc, argv);
-  for (auto & arg : args) {
-    if (arg == std::string("--use_multi_threaded_executor")) {
-      use_multi_threaded_executor = true;
+
+  std::string executor_type = "single-threaded";  // default
+  for (size_t i = 0; i < args.size(); ++i) {
+    if (args[i] == "--executor-type") {
+      if (i + 1 < args.size()) {
+        executor_type = args[i + 1];
+        break;
+      }
+    } else if (args[i] == "--use_multi_threaded_executor") { // backward compatibility
+      executor_type = "multi-threaded";
```

Comment:
it would be probably nice to add message for user, that this option is deprecated?